src/pages-and-resources/discussions/DiscussionsSettings.jsx

```diff
@@ -81,9 +81,9 @@ function DiscussionsSettings({ courseId, intl }) {
     );
   }
 
-  const handleConfigError = useCallback((isValid) => {
+  const handleConfigError = (isValid) => {
```

Comment:
`useCallback` should be kept here.  In general when a handler is passed to a sub-component, you want `useCallback` to ensure that it's not re-defined on re-render.  This helps keep the component tree efficient.

Reply:
I also feel the dependency list would include `[isValid]` rather than `[]` as in the case of the empty list the value will be memoized once and always returned.

Reply:
isValid is an argument to the callback - the dependency list includes variables from the lexical scope in which the callback is defined that are used inside it. So I think in this case it could just be []

Reply:
useCallback is breaking the test case on line 204 in DiscussionsSettings.test.jsx file. It's an async test case. When it tries to find an alert in the container it gets null value and fails the test. if we add some time delay for query the alert like 5 sec then alert is available in the container. if i remove useCallback it runs successfully. It's weird behaviour. I debugged it but i didn't find the reason behind this behaviour.